plenum/test/primary_selection/test_new_node_accepts_chosen_primary.py

```diff
@@ -47,6 +52,16 @@ def txnPoolNodeSet(txnPoolNodeSet, looper, client1, wallet1, client1Connected,
                  "viewNo: {}".format(txnPoolNodeSet[0].viewNo))
     sendReqsToNodesAndVerifySuffReplies(looper, wallet1, client1, numReqs=3)
 
+    logger.debug("Pool is ready, current viewNo: {}".format(txnPoolNodeSet[0].viewNo))
+
+    # TODO find out and fix why additional view change could happen
+    # because of degarded master. It's critical for current test to have
+    # view change completed for the time when new node is joining.
+    # Thus, disable master degradation check as it won't impact the case
+    # and guarantees necessary state.
+    for node in txnPoolNodeSet:
```

Comment:
Was it always like this, or it was introduced with the refactoring?

Reply:
I checked that against different past commits: seems it was always so and passed testing only because of specific environment where test worked as expected (no additional view changes happened). I added more comments for the tests in PR #480.